### ISSUE TITLE
Incorrect transparent surface ordering on map habitat

### ISSUE THREAD

**Issue by VReaperV** [DaemonEngine/Daemon#1079]
Some of the surfaces are drawn in an incorrect order in relation to the window:
![unvanquished_2024-04-13_104758_000](https://github.com/DaemonEngine/Daemon/assets/10687142/e167f79b-060a-4b22-8014-da21556c055b)
After the window is rendered:
![debug_tr_0](https://github.com/DaemonEngine/Daemon/assets/10687142/2492d190-eec2-451b-a5f6-5a74c4586abc)
After transparent objects behind the window are rendered:
![debug_tr_1](https://github.com/DaemonEngine/Daemon/assets/10687142/6af27097-c908-4c9b-95b2-34e3bfcd619c)
I believe it's an engine bug rather than a map bug since transparent surface should always be rendered back to front.

**Comment by illwieckz** [comment]
I agree it's likely an engine bug. I already fixed some of them long time ago but I will not be surprised if some are remaining, or came back.

It may be possible that map bug can produce this though, so we should double-check the map materials.

**Comment by slipher** [comment]
#1319 shows another example Here's a third:

![unvanquished_2024-10-24_020217_000](https://github.com/user-attachments/assets/4b991840-14df-43d7-ac1e-58c8f522413d)

I think the problem is we don't actually sort surfaces by distance at all. I don't see anything like that in `R_SortDrawSurfs`. It's always based on shaderSort_t+shader+entitynum+fog+lightmap, transparent or not.

**Comment by VReaperV** [comment]
> I think the problem is we don't actually sort surfaces by distance at all. I don't see anything like that in `R_SortDrawSurfs`. It's always based on shaderSort_t+shader+entitynum+fog+lightmap, transparent or not.

Apparently so. Perhaps just skipping translucent surfaces for this sort would be enough, since traversing the BSP returns surfaces essentialy ordered by distance.

**Comment by VReaperV** [comment]
There's a similar issue with projectiles (lcannon, prifle, blaster, maybe others) being rendered behind transparent geometry, even when they're actually in front of it. This wasn't the case before 0.55.

**Comment by slipher** [comment]
I got curious why the weapon effect vs. translucent map surface variant demonstrated in a Forlorn screenshot above wasn't seen more often before:
- An examination of glass shaders from `res-tremulous` shows that the majority of them use all additive or quasi-additive[1] stages. It doesn't make much sense for a glass to make everything behind it brighter, but that's what many of them do. (An additive lightmap stage does make some sense since with Quake 3 shaders there is no way to blend more than one texture at once, so it is impossible to do a standard alpha blend + lightmap.) Still, there are some shaders that don't that. For example nexus6 has glass on the wall opposite the human base which can exhibit the sorting bug. 
- In Unvanquished 0.53.0 we started to register some asset shaders before loading the map, due to the `LOAD_CONFIGS` step being moved earlier in Unvanquished 3dc34907ba7e3ca776ee8123f2c60bfd9f0cd3ac. So since 0.53.0 the weapon effects would always be rendered under windows instead of vice versa, because shader sorting depends on the order the shaders were registered in.

[1] The blend `gl_dst_color gl_src_alpha` was used on some lightmaps. Those don't have an alpha channel so it is equivalent to `gl_dst_color gl_one`. So the equation for e.g. the red channel is `red = (framebuffer's red * lightmap's red) + (1 * framebuffer's red) = (1 + lightmap's red) * framebuffer's red`. Like additive blending it can only make things brighter, but it is done via multiplication with a number >= 1.

**Comment by illwieckz** [comment]
> * In Unvanquished 0.53.0 we started to register some asset shaders before loading the map, due to the `LOAD_CONFIGS` step being moved earlier in Unvanquished 3dc34907ba7e3ca776ee8123f2c60bfd9f0cd3ac. So since 0.53.0 the weapon effects would always be rendered under windows instead of vice versa, because shader sorting depends on the order the shaders were registered in.

So, does this improve things a bit?

- https://github.com/Unvanquished/Unvanquished/pull/3312
